Library/Formula/pcs.rb

```diff
@@ -0,0 +1,38 @@
+class Pcs < Formula
+  desc "The terminal utility for Baidu Network Disk."
+  homepage "https://github.com/GangZhuo/BaiduPCS/"
+  url "https://github.com/GangZhuo/BaiduPCS/archive/0.2.3.tar.gz"
+  sha256 "e0a72e41560df3f1499cf5baf29404465d34b14a9bff82095335dd29f0a0216f"
+
+  head "https://github.com/GangZhuo/BaiduPCS.git"
+
+  depends_on "openssl"
+
+  def install
+    system "make", "clean"
```

Comment:
I found if the repository is dirty, `brew install --HEAD pcs` won't generate a new version. So, I think it's better to run `make clean` before `make`.

Steps:
  0. brew uninstall pcs (v0.2.3)
  1. check out the version 0.2.2 and make it in /Library/Caches/Homebrew/pcs--git
  2. brew install --HEAD pcs, this will make the repository check out latest code.
  3. pcs is v0.2.2!!!
  4. brew uninstall pcs
  5. brew install --HEAD pcs
  6. pcs is still v0.2.2
  7. run `make` in /Library/Caches/Homebrew/pcs--git
  8. reinstall 'pcs' using brew
  9. pcs is v0.2.3 now.

Let me know if I made mistake. Thank you.

here the log:
<pre>
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » command -V pcs                                                  255 ↵
> pcs is /usr/local/bin/pcs
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » pcs version
> pcs v0.2.3 (API v1.1.1)pcs
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » brew uninstall pcs
> Uninstalling /usr/local/Cellar/pcs/HEAD... (4 files, 182.8K)
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » command -V pcs
> pcs not found
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » gco v0.2.2      2>/dev/null
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±384184c » cp /tmp/makefile.0.2.3 makefile # there is a bug, use makefile of v0.2.3 instead
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±384184c⚡ » gd | cat
> diff --git a/makefile b/makefile
> index 7b44a32..1dc66d3 100644
> --- a/makefile
> +++ b/makefile
> @@ -14,7 +14,7 @@ CYGWIN_CCFLAGS =
>  endif
> 
>  ifeq ($(LC_OS_NAME), darwin)
> -APPLE_CCFLAGS = -largp
> +APPLE_CCFLAGS = -I/usr/local/opt/openssl/include
>  else
>  APPLE_CCFLAGS =
>  endif
> @@ -26,7 +26,7 @@ else
>  CCFLAGS:=-g -D_FILE_OFFSET_BITS=64 -DDEBUG -D_DEBUG
>  endif
> 
> -PCS_CCFLAGS = -fPIC $(CCFLAGS)
> +PCS_CCFLAGS = -fPIC $(CCFLAGS) $(CYGWIN_CCFLAGS) $(APPLE_CCFLAGS)
> 
>  all: bin/libpcs.a bin/pcs
> 
> @@ -102,7 +102,7 @@ uninstall4-dev:
> 
>  .PHONY : clean
>  clean :
> -	-rm ./bin/*.o ./bin/*.so ./bin/libpcs.a ./bin/pcs
> +	-rm -f ./bin/*.o ./bin/*.so ./bin/libpcs.a ./bin/pcs
> 
>  .PHONY : pre
>  pre :
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±384184c⚡ » ll bin/pcs
> ls: cannot access 'bin/pcs': No such file or directory
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±384184c⚡ » make 1>/dev/null 2>&1                                           2 ↵
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±384184c⚡ » bin/pcs version
> pcs v0.2.2 (API v1.1.1)pcs
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±384184c⚡ » brew install --verbose --HEAD pcs
> ==> Cloning https://github.com/GangZhuo/BaiduPCS.git
> Updating /Library/Caches/Homebrew/pcs--git
> git config remote.origin.url https://github.com/GangZhuo/BaiduPCS.git
> git config remote.origin.fetch +refs/heads/master:refs/remotes/origin/master
> git fetch origin
> ==> Checking out branch master
> git checkout -f master --
> Previous HEAD position was 384184c... bump v0.2.2
> Switched to branch 'master'
> Your branch is up-to-date with 'origin/master'.
> git reset --hard origin/master
> HEAD is now at 408e175 fix bugs
> ==> make
> makefile:23: "Use 'make ver=debug' to build for gdb debug."
> mkdir -p bin/
> ar crv bin/libpcs.a bin/cJSON.o bin/pcs.o bin/pcs_fileinfo.o bin/pcs_http.o bin/pcs_mem.o bin/pcs_pan_api_resinfo.o bin/pcs_slist.o bin/pcs_utils.o
> r - bin/cJSON.o
> r - bin/pcs.o
> r - bin/pcs_fileinfo.o
> r - bin/pcs_http.o
> r - bin/pcs_mem.o
> r - bin/pcs_pan_api_resinfo.o
> r - bin/pcs_slist.o
> r - bin/pcs_utils.o
> clang -o bin/pcs bin/shell_arg.o bin/shell.o bin/dir.o bin/rb_tree_misc.o bin/rb_tree_stack.o bin/red_black_tree.o bin/shell_utils.o bin/hashtable.o -D_FILE_OFFSET_BITS=64  -I/usr/local/opt/openssl/include -L./bin -lpcs -lm -lcurl -lssl -lcrypto -lpthread
> ==> Cleaning
> ==> Finishing up
> ln -s ../Cellar/pcs/HEAD/bin/pcs pcs
> ==> Caveats
> The pcs is client of baidu net disk. It supplied many functions,
> which can manage baidu net disk on terminal, such as ls, cp, rm,
> mv, rename, download, upload, search and so on.
> 
> The pcs provided AES encryption, which can protected your data.
> 
> The pcs is open source, and published on MIT.
> Please see https://github.com/GangZhuo/baidupcs.
> 
> To login with your Baidu account, run:
>   pcs login
> 
> To get help, run:
>   pcs help
> ==> Summary
> 🍺  /usr/local/Cellar/pcs/HEAD: 4 files, 266.1K, built in 2 seconds
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » command -V pcs
> pcs is /usr/local/bin/pcs
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » pcs version
> pcs v0.2.2 (API v1.1.1)pcs
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » brew uninstall pcs
> Uninstalling /usr/local/Cellar/pcs/HEAD... (4 files, 266.1K)
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » brew install --verbose --HEAD pcs
> ==> Cloning https://github.com/GangZhuo/BaiduPCS.git
> Updating /Library/Caches/Homebrew/pcs--git
> git config remote.origin.url https://github.com/GangZhuo/BaiduPCS.git
> git config remote.origin.fetch +refs/heads/master:refs/remotes/origin/master
> git fetch origin
> ==> Checking out branch master
> git checkout -f master --
> Already on 'master'
> Your branch is up-to-date with 'origin/master'.
> git reset --hard origin/master
> HEAD is now at 408e175 fix bugs
> ==> make
> makefile:23: "Use 'make ver=debug' to build for gdb debug."
> mkdir -p bin/
> ar crv bin/libpcs.a bin/cJSON.o bin/pcs.o bin/pcs_fileinfo.o bin/pcs_http.o bin/pcs_mem.o bin/pcs_pan_api_resinfo.o bin/pcs_slist.o bin/pcs_utils.o
> r - bin/cJSON.o
> r - bin/pcs.o
> r - bin/pcs_fileinfo.o
> r - bin/pcs_http.o
> r - bin/pcs_mem.o
> r - bin/pcs_pan_api_resinfo.o
> r - bin/pcs_slist.o
> r - bin/pcs_utils.o
> clang -o bin/pcs bin/shell_arg.o bin/shell.o bin/dir.o bin/rb_tree_misc.o bin/rb_tree_stack.o bin/red_black_tree.o bin/shell_utils.o bin/hashtable.o -D_FILE_OFFSET_BITS=64  -I/usr/local/opt/openssl/include -L./bin -lpcs -lm -lcurl -lssl -lcrypto -lpthread
> ==> Cleaning
> ==> Finishing up
> ln -s ../Cellar/pcs/HEAD/bin/pcs pcs
> ==> Caveats
> The pcs is client of baidu net disk. It supplied many functions,
> which can manage baidu net disk on terminal, such as ls, cp, rm,
> mv, rename, download, upload, search and so on.
> 
> The pcs provided AES encryption, which can protected your data.
> 
> The pcs is open source, and published on MIT.
> Please see https://github.com/GangZhuo/baidupcs.
> 
> To login with your Baidu account, run:
>   pcs login
> 
> To get help, run:
>   pcs help
> ==> Summary
> 🍺  /usr/local/Cellar/pcs/HEAD: 4 files, 266.1K, built in 2 seconds
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » command -V pcs; pcs version
> pcs is /usr/local/bin/pcs
> pcs v0.2.2 (API v1.1.1)pcs
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » make 1>/dev/null 2>&1
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » bin/pcs version
> pcs v0.2.3 (API v1.1.1)pcs
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » brew uninstall pcs
> Uninstalling /usr/local/Cellar/pcs/HEAD... (4 files, 266.1K)
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » brew install --verbose --HEAD pcs
> ==> Cloning https://github.com/GangZhuo/BaiduPCS.git
> Updating /Library/Caches/Homebrew/pcs--git
> git config remote.origin.url https://github.com/GangZhuo/BaiduPCS.git
> git config remote.origin.fetch +refs/heads/master:refs/remotes/origin/master
> git fetch origin
> ==> Checking out branch master
> git checkout -f master --
> Already on 'master'
> Your branch is up-to-date with 'origin/master'.
> git reset --hard origin/master
> HEAD is now at 408e175 fix bugs
> ==> make
> makefile:23: "Use 'make ver=debug' to build for gdb debug."
> mkdir -p bin/
> ar crv bin/libpcs.a bin/cJSON.o bin/pcs.o bin/pcs_fileinfo.o bin/pcs_http.o bin/pcs_mem.o bin/pcs_pan_api_resinfo.o bin/pcs_slist.o bin/pcs_utils.o
> r - bin/cJSON.o
> r - bin/pcs.o
> r - bin/pcs_fileinfo.o
> r - bin/pcs_http.o
> r - bin/pcs_mem.o
> r - bin/pcs_pan_api_resinfo.o
> r - bin/pcs_slist.o
> r - bin/pcs_utils.o
> clang -o bin/pcs bin/shell_arg.o bin/shell.o bin/dir.o bin/rb_tree_misc.o bin/rb_tree_stack.o bin/red_black_tree.o bin/shell_utils.o bin/hashtable.o -D_FILE_OFFSET_BITS=64  -I/usr/local/opt/openssl/include -L./bin -lpcs -lm -lcurl -lssl -lcrypto -lpthread
> ==> Cleaning
> ==> Finishing up
> ln -s ../Cellar/pcs/HEAD/bin/pcs pcs
> ==> Caveats
> The pcs is client of baidu net disk. It supplied many functions,
> which can manage baidu net disk on terminal, such as ls, cp, rm,
> mv, rename, download, upload, search and so on.
> 
> The pcs provided AES encryption, which can protected your data.
> 
> The pcs is open source, and published on MIT.
> Please see https://github.com/GangZhuo/baidupcs.
> 
> To login with your Baidu account, run:
>   pcs login
> 
> To get help, run:
>   pcs help
> ==> Summary
> 🍺  /usr/local/Cellar/pcs/HEAD: 4 files, 270.3K, built in 3 seconds
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » command -V pcs; pcs version
> pcs is /usr/local/bin/pcs
> pcs v0.2.3 (API v1.1.1)pcs
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master »chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » command -V pcs                                                  255 ↵
> pcs is /usr/local/bin/pcs
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » pcs version
> pcs v0.2.3 (API v1.1.1)pcs
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » brew uninstall pcs
> Uninstalling /usr/local/Cellar/pcs/HEAD... (4 files, 182.8K)
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » command -V pcs
> pcs not found
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » gco v0.2.2 2>/dev/null                                            1 ↵
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±384184c » cp /tmp/makefile.0.2.3 makefile
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±384184c⚡ » gd | cat
> diff --git a/makefile b/makefile
> index 7b44a32..1dc66d3 100644
> --- a/makefile
> +++ b/makefile
> @@ -14,7 +14,7 @@ CYGWIN_CCFLAGS =
>  endif
> 
>  ifeq ($(LC_OS_NAME), darwin)
> -APPLE_CCFLAGS = -largp
> +APPLE_CCFLAGS = -I/usr/local/opt/openssl/include
>  else
>  APPLE_CCFLAGS =
>  endif
> @@ -26,7 +26,7 @@ else
>  CCFLAGS:=-g -D_FILE_OFFSET_BITS=64 -DDEBUG -D_DEBUG
>  endif
> 
> -PCS_CCFLAGS = -fPIC $(CCFLAGS)
> +PCS_CCFLAGS = -fPIC $(CCFLAGS) $(CYGWIN_CCFLAGS) $(APPLE_CCFLAGS)
> 
>  all: bin/libpcs.a bin/pcs
> 
> @@ -102,7 +102,7 @@ uninstall4-dev:
> 
>  .PHONY : clean
>  clean :
> -	-rm ./bin/*.o ./bin/*.so ./bin/libpcs.a ./bin/pcs
> +	-rm -f ./bin/*.o ./bin/*.so ./bin/libpcs.a ./bin/pcs
> 
>  .PHONY : pre
>  pre :
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±384184c⚡ » ll bin/pcs
> ls: cannot access 'bin/pcs': No such file or directory
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±384184c⚡ » make 1>/dev/null 2>&1                                           2 ↵
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±384184c⚡ » bin/pcs version
> pcs v0.2.2 (API v1.1.1)pcs
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±384184c⚡ » brew install --verbose --HEAD pcs
> ==> Cloning https://github.com/GangZhuo/BaiduPCS.git
> Updating /Library/Caches/Homebrew/pcs--git
> git config remote.origin.url https://github.com/GangZhuo/BaiduPCS.git
> git config remote.origin.fetch +refs/heads/master:refs/remotes/origin/master
> git fetch origin
> ==> Checking out branch master
> git checkout -f master --
> Previous HEAD position was 384184c... bump v0.2.2
> Switched to branch 'master'
> Your branch is up-to-date with 'origin/master'.
> git reset --hard origin/master
> HEAD is now at 408e175 fix bugs
> ==> make
> makefile:23: "Use 'make ver=debug' to build for gdb debug."
> mkdir -p bin/
> ar crv bin/libpcs.a bin/cJSON.o bin/pcs.o bin/pcs_fileinfo.o bin/pcs_http.o bin/pcs_mem.o bin/pcs_pan_api_resinfo.o bin/pcs_slist.o bin/pcs_utils.o
> r - bin/cJSON.o
> r - bin/pcs.o
> r - bin/pcs_fileinfo.o
> r - bin/pcs_http.o
> r - bin/pcs_mem.o
> r - bin/pcs_pan_api_resinfo.o
> r - bin/pcs_slist.o
> r - bin/pcs_utils.o
> clang -o bin/pcs bin/shell_arg.o bin/shell.o bin/dir.o bin/rb_tree_misc.o bin/rb_tree_stack.o bin/red_black_tree.o bin/shell_utils.o bin/hashtable.o -D_FILE_OFFSET_BITS=64  -I/usr/local/opt/openssl/include -L./bin -lpcs -lm -lcurl -lssl -lcrypto -lpthread
> ==> Cleaning
> ==> Finishing up
> ln -s ../Cellar/pcs/HEAD/bin/pcs pcs
> ==> Caveats
> The pcs is client of baidu net disk. It supplied many functions,
> which can manage baidu net disk on terminal, such as ls, cp, rm,
> mv, rename, download, upload, search and so on.
> 
> The pcs provided AES encryption, which can protected your data.
> 
> The pcs is open source, and published on MIT.
> Please see https://github.com/GangZhuo/baidupcs.
> 
> To login with your Baidu account, run:
>   pcs login
> 
> To get help, run:
>   pcs help
> ==> Summary
> 🍺  /usr/local/Cellar/pcs/HEAD: 4 files, 266.1K, built in 2 seconds
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » command -V pcs
> pcs is /usr/local/bin/pcs
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » pcs version
> pcs v0.2.2 (API v1.1.1)pcs
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » brew uninstall pcs
> Uninstalling /usr/local/Cellar/pcs/HEAD... (4 files, 266.1K)
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » brew install --verbose --HEAD pcs
> ==> Cloning https://github.com/GangZhuo/BaiduPCS.git
> Updating /Library/Caches/Homebrew/pcs--git
> git config remote.origin.url https://github.com/GangZhuo/BaiduPCS.git
> git config remote.origin.fetch +refs/heads/master:refs/remotes/origin/master
> git fetch origin
> ==> Checking out branch master
> git checkout -f master --
> Already on 'master'
> Your branch is up-to-date with 'origin/master'.
> git reset --hard origin/master
> HEAD is now at 408e175 fix bugs
> ==> make
> makefile:23: "Use 'make ver=debug' to build for gdb debug."
> mkdir -p bin/
> ar crv bin/libpcs.a bin/cJSON.o bin/pcs.o bin/pcs_fileinfo.o bin/pcs_http.o bin/pcs_mem.o bin/pcs_pan_api_resinfo.o bin/pcs_slist.o bin/pcs_utils.o
> r - bin/cJSON.o
> r - bin/pcs.o
> r - bin/pcs_fileinfo.o
> r - bin/pcs_http.o
> r - bin/pcs_mem.o
> r - bin/pcs_pan_api_resinfo.o
> r - bin/pcs_slist.o
> r - bin/pcs_utils.o
> clang -o bin/pcs bin/shell_arg.o bin/shell.o bin/dir.o bin/rb_tree_misc.o bin/rb_tree_stack.o bin/red_black_tree.o bin/shell_utils.o bin/hashtable.o -D_FILE_OFFSET_BITS=64  -I/usr/local/opt/openssl/include -L./bin -lpcs -lm -lcurl -lssl -lcrypto -lpthread
> ==> Cleaning
> ==> Finishing up
> ln -s ../Cellar/pcs/HEAD/bin/pcs pcs
> ==> Caveats
> The pcs is client of baidu net disk. It supplied many functions,
> which can manage baidu net disk on terminal, such as ls, cp, rm,
> mv, rename, download, upload, search and so on.
> 
> The pcs provided AES encryption, which can protected your data.
> 
> The pcs is open source, and published on MIT.
> Please see https://github.com/GangZhuo/baidupcs.
> 
> To login with your Baidu account, run:
>   pcs login
> 
> To get help, run:
>   pcs help
> ==> Summary
> 🍺  /usr/local/Cellar/pcs/HEAD: 4 files, 266.1K, built in 2 seconds
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » command -V pcs; pcs version
> pcs is /usr/local/bin/pcs
> pcs v0.2.2 (API v1.1.1)pcs
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » make 1>/dev/null 2>&1
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » bin/pcs version
> pcs v0.2.3 (API v1.1.1)pcs
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » brew uninstall pcs
> Uninstalling /usr/local/Cellar/pcs/HEAD... (4 files, 266.1K)
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » brew install --verbose --HEAD pcs
> ==> Cloning https://github.com/GangZhuo/BaiduPCS.git
> Updating /Library/Caches/Homebrew/pcs--git
> git config remote.origin.url https://github.com/GangZhuo/BaiduPCS.git
> git config remote.origin.fetch +refs/heads/master:refs/remotes/origin/master
> git fetch origin
> ==> Checking out branch master
> git checkout -f master --
> Already on 'master'
> Your branch is up-to-date with 'origin/master'.
> git reset --hard origin/master
> HEAD is now at 408e175 fix bugs
> ==> make
> makefile:23: "Use 'make ver=debug' to build for gdb debug."
> mkdir -p bin/
> ar crv bin/libpcs.a bin/cJSON.o bin/pcs.o bin/pcs_fileinfo.o bin/pcs_http.o bin/pcs_mem.o bin/pcs_pan_api_resinfo.o bin/pcs_slist.o bin/pcs_utils.o
> r - bin/cJSON.o
> r - bin/pcs.o
> r - bin/pcs_fileinfo.o
> r - bin/pcs_http.o
> r - bin/pcs_mem.o
> r - bin/pcs_pan_api_resinfo.o
> r - bin/pcs_slist.o
> r - bin/pcs_utils.o
> clang -o bin/pcs bin/shell_arg.o bin/shell.o bin/dir.o bin/rb_tree_misc.o bin/rb_tree_stack.o bin/red_black_tree.o bin/shell_utils.o bin/hashtable.o -D_FILE_OFFSET_BITS=64  -I/usr/local/opt/openssl/include -L./bin -lpcs -lm -lcurl -lssl -lcrypto -lpthread
> ==> Cleaning
> ==> Finishing up
> ln -s ../Cellar/pcs/HEAD/bin/pcs pcs
> ==> Caveats
> The pcs is client of baidu net disk. It supplied many functions,
> which can manage baidu net disk on terminal, such as ls, cp, rm,
> mv, rename, download, upload, search and so on.
> 
> The pcs provided AES encryption, which can protected your data.
> 
> The pcs is open source, and published on MIT.
> Please see https://github.com/GangZhuo/baidupcs.
> 
> To login with your Baidu account, run:
>   pcs login
> 
> To get help, run:
>   pcs help
> ==> Summary
> 🍺  /usr/local/Cellar/pcs/HEAD: 4 files, 270.3K, built in 3 seconds
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master » command -V pcs; pcs version
> pcs is /usr/local/bin/pcs
> pcs v0.2.3 (API v1.1.1)pcs
> chaifeng@chairmbp /Library/Caches/Homebrew/pcs--git ±master »
</pre>